### PR TITLE
Feature/35 logout from session 

### LINKED ISSUE #35: Logout from session
## TODO/DONE

- [x] Remove sessions from db and state
- [x] Unit tests

### PR DESCRIPTION
**Pull request by ifirmawan**
## TODO/DONE

- [x] Remove sessions from db and state
- [x] Unit tests

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204955479635210

### CODE REVIEW

File: app/src/pages/Settings/Form.js
Comment (dedenbangkit) [inline app/src/pages/Settings/Form.js:1]: @ifirmawan please change the name to SettingsForm instead Form.js

File: app/src/components/LogoutButton.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  const handleError = () => {
+    if (Platform.OS === 'android') {
+      ToastAndroid.show('Logout Failed: Please try again later', ToastAndroid.LONG);
+    }
+  };
+
+  const handleYesPress = () => {
+    const table = 'sessions';
+    const clearQuery = query.clear(table);
+    console.log('clear', clearQuery);
```
Comment (dedenbangkit) [inline app/src/components/LogoutButton.js]: remove console log

File: app/src/components/LogoutButton.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  const [loading, setLoading] = useState(false);
+  const navigation = useNavigation();
+
+  const handleNoPress = () => {
+    setVisible(false);
+  };
+
+  const handleError = () => {
+    if (Platform.OS === 'android') {
+      ToastAndroid.show('Logout Failed: Please try again later', ToastAndroid.LONG);
```
Comment (dedenbangkit) [inline app/src/components/LogoutButton.js]: why do we even need this?

File: app/src/components/LogoutButton.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    }
+  };
+
+  const handleYesPress = () => {
+    const table = 'sessions';
+    const clearQuery = query.clear(table);
```
Comment (dedenbangkit) [inline app/src/components/LogoutButton.js]: Please create truncate query instead of just clearing the session.

**Review by dedenbangkit — CHANGES_REQUESTED** [review]

File: app/src/pages/Settings/Form.js
Comment (ifirmawan) [reply app/src/pages/Settings/Form.js:1]: Okay mas, on it

File: app/src/components/LogoutButton.js (hunk 3 above)
Comment (ifirmawan) [reply app/src/components/LogoutButton.js]: Hi mas @dedenbangkit actually clear is truncate query
https://github.com/akvo/nmis-mobile/blob/main/app/src/database/query.js#L25-L27

Comment (dedenbangkit) [reply app/src/components/LogoutButton.js]: Sorry, I mean the truncate should support multiple tables

**Review by dedenbangkit — APPROVED** [review]

**Pull request merged**